libs/designsystem/src/lib/components/list/helpers/list-helper.spec.ts

```diff
@@ -106,4 +106,167 @@ describe('list helper', () => {
       expect(component.isLoadOnDemandEnabled).toBeTruthy();
     });
   });
+
+  describe('function: groupSections', () => {
+    it('should section list by callback function', () => {
+      const callback = (item: any) => {
+        return item.title;
+      };
+      const collection = [
+        {
+          title: 'section 1',
+          value: 1,
+        },
+        {
+          title: 'section 2',
+          value: 2,
+        },
+        {
+          title: 'section 3',
+          value: 3,
+        },
+      ];
+
+      const result = listHelper.groupSections(collection, callback);
+
+      expect(result.length).toBe(3);
+      result.forEach((section: any, index: number) => {
+        expect(section.items.length).toBe(1);
+        expect(section.items[0]).toEqual(collection[index]);
+      });
+    });
+
+    it('should order sections by alphabetical order', () => {
+      const callback = (item: any) => {
+        return item.title;
+      };
+      const collection = [
+        {
+          title: 'section 202',
+          value: 1,
+        },
+        {
+          title: 'section 1',
+          value: 2,
+        },
+        {
+          title: 'section 44',
+          value: 3,
+        },
+      ];
+
+      const result = listHelper.groupSections(collection, callback);
+      expect(result[0].name).toEqual('section 1');
+      expect(result[1].name).toEqual('section 202');
+      expect(result[2].name).toEqual('section 44');
+    });
+  });
+
+  describe('function: groupStandAloneItems', () => {
+    it('should divide items into sublists based on "standAloneProperty"', () => {
+      const standAloneProperty = 'isStandAlone';
+      const collection = [
+        {
+          title: 'Item 1',
+          value: 1,
+        },
+        {
+          title: 'Item 2',
+          value: 2,
+        },
+        {
+          title: 'Item 3',
+          value: 3,
+          isStandAlone: true,
+        },
+        {
+          title: 'Item 4',
+          value: 4,
+        },
+        {
+          title: 'Item 5',
+          value: 5,
+        },
+      ];
+
+      const result = listHelper.groupStandAloneItems(collection, standAloneProperty);
+
+      expect(result[0]).toEqual({
+        items: [
+          {
+            title: 'Item 1',
+            value: 1,
+          },
+          {
+            title: 'Item 2',
+            value: 2,
+          },
+        ],
+      });
+      expect(result[1]).toEqual({
+        items: [
+          {
+            title: 'Item 3',
+            value: 3,
+            isStandAlone: true,
+          },
+        ],
+      });
+      expect(result[2]).toEqual({
+        items: [
+          {
+            title: 'Item 4',
+            value: 4,
+          },
+          {
+            title: 'Item 5',
+            value: 5,
+          },
+        ],
+      });
+    });
+  });
+
+  describe('function: groupSectionsWithStandAloneItems', () => {
+    fit('should create sections with sublists"', () => {
```

Comment:
Couldn't help myself and had a little sneak peek on this PR.
Remember to remove this focussed test! 😄

Reply:
There's also some exciting stuff in here 😄 Ahh that one slipped 😂 Nice catch!